File: Backend/src/main/java/com/soen390/flightcrew/model/PointOfInterest.java
package com.soen390.flightcrew.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class PointOfInterest {

    @JsonProperty("Name")
    private String name;

    @JsonProperty("Category")
    private String category;

    @JsonProperty("Campus")
    private String campus;

    @JsonProperty("Address")
    private String address;

    @JsonProperty("Latitude")
    private Double latitude;

    @JsonProperty("Longitude")
    private Double longitude;

    @JsonProperty("Description")
    private String description;

    @JsonProperty("Google_Place_Info")
    private GoogleGeocodeResponse.PrimaryPlace googlePlaceInfo;
}

File: mobile/src/types/PointOfInterest.ts
export interface PointOfInterest {
  name: string;
  category: PoiCategory;
  campus: string;
  address: string;
  latitude: number;
  longitude: number;
  description: string;
  googlePlaceInfo?: {
    displayName?: { text: string };
  };
}

export type PoiCategory =
  | "cafe"
  | "restaurant"
  | "pharmacy"
  | "bar"
  | "grocery";
